reascripts/ReaSpeech/source/ASRActions.lua

```diff
@@ -0,0 +1,176 @@
+--[[
+
+  ASRActions.lua - Actions for ASR plugin
+
+]]--
+
+ASRActions = Polo {
+  new = function(plugin)
+    return {
+      plugin = plugin
+    }
+  end,
+}
+
+function ASRActions:init()
+  assert(self.plugin, 'ASRActions: plugin is required')
+
+  Logging.init(self, 'ASRActions')
+
+  self.disabler = ReaUtil.disabler(ctx)
+end
+
+function ASRActions:actions()
+  return {
+    self:selected_tracks_button(),
+    self:selected_items_button(),
+    self:all_items_button(),
+  }
+end
```

Comment:
these methods will watch for changes in the selected item/track count and return/cache new button instances to reflect new values.